core/src/main/java/org/geysermc/geyser/translator/protocol/java/inventory/JavaMerchantOffersTranslator.java

```diff
@@ -57,6 +57,18 @@ public void translate(GeyserSession session, ClientboundMerchantOffersPacket pac
             return;
         }
 
+        // No previous inventory was closed -> no need of queuing the merchant inventory
+        if (!openInventory.isPending()) {
+            openMerchant(session, packet, merchantInventory);
+            return;
+        }
+
+        // The inventory is declared as pending due to previous closing inventory -> leads to an incorrect order of execution
+        // Handled in BedrockContainerCloseTranslator
+        merchantInventory.setPendingOffersPacket(packet);
+    }
+
+    public static void openMerchant(GeyserSession session, ClientboundMerchantOffersPacket packet, MerchantContainer merchantInventory) {
```

Comment:
Not really a fan of this being in the packet translator class, may be better to move this method to `MerchantContainer`.